Marlin/power.cpp

```diff
@@ -0,0 +1,88 @@
+//
+// Created by christopher on 20.02.17.
+//
+
+#include "power.h"
+#include "temperature.h"
+#include "stepper_indirection.h"
+#include "Marlin.h"
+#include "MarlinConfig.h"
+#include "ultralcd.h"
+
+void Power::check() {
+  bool powerNeeded = false;
+  static millis_t lastPowerOn = 0;
+  static millis_t nextPowerCheck = 0;
+  millis_t ms = millis();
+  if (ELAPSED(ms, nextPowerCheck)) {
+    nextPowerCheck = ms + 2500UL;
+
+#if ENABLED(FAN_NEEDS_POWER)
+    for (uint8_t i = 0; i < FAN_COUNT; i++) {
+      if (fanSpeeds[i] > 0) {
+        powerNeeded = true;
```

Comment:
It would be better to have a `is_power_needed` function (inline, perhaps) which has `return true` here. (Thus skipping all future tests.)